src/util-inl.h

```diff
@@ -229,30 +230,85 @@ bool StringEqualNoCaseN(const char* a, const char* b, size_t length) {
   return true;
 }
 
+// clang
+#ifndef __has_builtin
+#define __has_builtin(x) (0)
+#endif
+
+template<typename T>
+T MultiplyWithOverflowCheck(T a, T b) {
+  T ret;
+
+#if NODE_GNUC_AT_LEAST(5, 0, 0) || __has_builtin(__builtin_mul_overflow)
+  CHECK(!__builtin_mul_overflow(a, b, &ret));
```

Comment:
I'm not a big fan of using compiler builtins when plain C++ suffices.  Is there a dramatic, real world performance difference?  If not, I would suggest dropping this.